pkg/ui/config/config.go

```diff
@@ -0,0 +1,159 @@
+	}
+}
+
+// Config stores the configuration for gcs bucket.
+type gcsConfig struct {
+	Type   client.ObjProvider
+	Config struct {
+		Bucket         string `yaml:"bucket"`
+		ServiceAccount string `yaml:"service_account"`
+	}
+}
+
+// Config encapsulates the necessary config values to instantiate an cos client.
+type cosConfig struct {
+	Type   client.ObjProvider
+	Config struct {
+		Bucket    string        `yaml:"bucket"`
+		Region    string        `yaml:"region"`
+		AppId     string        `yaml:"app_id"`
+		SecretKey config.Secret `yaml:"secret_key"`
+		SecretId  string        `yaml:"secret_id"`
+	}
+}
+
+// Config Azure storage configuration.
+type azureConfig struct {
+	Type   client.ObjProvider
+	Config struct {
+		StorageAccountName string        `yaml:"storage_account"`
+		StorageAccountKey  config.Secret `yaml:"storage_account_key"`
+		ContainerName      string        `yaml:"container"`
+		Endpoint           string        `yaml:"endpoint"`
+		MaxRetries         int           `yaml:"max_retries"`
+	}
+}
+
+// Config Swift storage configuration.
+type swiftConfig struct {
+	Type   client.ObjProvider
+	Config struct {
+		AuthUrl           string        `yaml:"auth_url"`
+		Username          string        `yaml:"username"`
+		UserDomainName    string        `yaml:"user_domain_name"`
+		UserDomainID      string        `yaml:"user_domain_id"`
+		UserId            string        `yaml:"user_id"`
+		Password          config.Secret `yaml:"password"`
+		DomainId          string        `yaml:"domain_id"`
+		DomainName        string        `yaml:"domain_name"`
+		ProjectID         string        `yaml:"project_id"`
+		ProjectName       string        `yaml:"project_name"`
+		ProjectDomainID   string        `yaml:"project_domain_id"`
+		ProjectDomainName string        `yaml:"project_domain_name"`
+		RegionName        string        `yaml:"region_name"`
+		ContainerName     string        `yaml:"container_name"`
+	}
+}
+
+// Config Swift storage configuration.
+type s3Config struct {
+	Type   client.ObjProvider
+	Config struct {
+		Bucket             string            `yaml:"bucket"`
+		Endpoint           string            `yaml:"endpoint"`
+		Region             string            `yaml:"region"`
+		AccessKey          string            `yaml:"access_key"`
+		Insecure           bool              `yaml:"insecure"`
+		SignatureV2        bool              `yaml:"signature_version2"`
+		SecretKey          config.Secret     `yaml:"secret_key"`
+		PutUserMetadata    map[string]string `yaml:"put_user_metadata"`
+		HTTPConfig         s3.HTTPConfig     `yaml:"http_config"`
+		TraceConfig        s3.TraceConfig    `yaml:"trace"`
+		ListObjectsVersion string            `yaml:"list_objects_version"`
+		// PartSize used for multipart upload. Only used if uploaded object size is known and larger than configured PartSize.
+		PartSize  uint64       `yaml:"part_size"`
+		SSEConfig s3.SSEConfig `yaml:"sse_config"`
+	}
+}
```

Comment:
We can use the `Config` structs already defined in their respective packages. For example, instead of redefining `gcsConfig` here, you can import `github.com/thanos-io/thanos/pkg/objstore/gcs` and use `gcs.Config` directly in the ReplaceSecret` function, just like you used `client.BucketConfig`.

Reply:
We should try to reuse the existing definitions if we could. In case we fail we can fallback to redefining all of these.